### PR TITLE
Request line helper object

### PR DESCRIPTION
**Pull request by g4s8**
#12 - added request line helper object with unit tests

### CODE REVIEW

**Comment by g4s8** [comment]
@0crat elect

**Comment by 0crat** [comment]
This pull request [#13](https://github.com/artipie/http/issues/13) is assigned to @iakunin[/z](https://www.0crat.com/u/iakunin), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/61048309-1897-4f67-97e0-18ae980aab3b); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/http/rq/RequestLine.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    }
+
+    /**
+     * Part of request line.
+     * @param idx Part index
+     * @return Part string
+     * @throws IOException In case of IO error
+     */
+    private String part(final int idx) throws IOException {
+        return new IoChecked<>(this.line::asString).value().split("\\s")[idx];
```
Comment (iakunin) [inline src/main/java/com/artipie/http/rq/RequestLine.java]: @g4s8 maybe it's better to rewrite it using `ItemAt`?

```
return new IoChecked<>(
    () -> new ItemAt<>(
        idx,
        new Split(
            this.line,
            "\\s"
        )
    ).value().asString()
).value();
```

Or even simpler if we're able to change exception in signature to just `Exception`:
```
new ItemAt<>(
    idx,
    new Split(
        this.line,
        "\\s"
    )
).value().asString();
```

Another option is to change method signature to `private Text part(final int idx) throws IOException`. That'll give us opportunity to write even more concise:
```
new IoChecked<>(
    new ItemAt<>(
        idx,
        new Split(
            this.line,
            "\\s"
        )
    )
).value();
```

Comment (g4s8) [reply src/main/java/com/artipie/http/rq/RequestLine.java]: @iakunin it looks to verbose for me, see:
```java
return new IoChecked<>(
    () -> new ItemAt<>(idx, new Split(this.line, "\\s")).value().asString()
).value();
```
vs:
```java
return new IoChecked<>(this.line::asString).value().split("\\s")[idx];
```
The problem of Cactoos version here is verbose exception handling for code which can't throw checked exceptions.

Comment (iakunin) [reply src/main/java/com/artipie/http/rq/RequestLine.java]: @g4s8 got it

Comment (iakunin) [inline src/main/java/com/artipie/http/rq/RequestLine.java]: @g4s8 here can be `ArrayIndexOutOfBoundsException`: should we handle this case?

Comment (g4s8) [reply src/main/java/com/artipie/http/rq/RequestLine.java]: @iakunin I'd postpone this case for later, it's a corner case if status line is broken and can't be split into 3 parts. It's possible only if HTTP server accepts invalid HTTP request message.

Comment (iakunin) [reply src/main/java/com/artipie/http/rq/RequestLine.java]: @g4s8 maybe it's better to leave a puzzle for that?

Comment (g4s8) [reply src/main/java/com/artipie/http/rq/RequestLine.java]: @iakunin done: https://github.com/artipie/http/pull/13/commits/4a5c8097b1523dac325dc4fb8600bf4c51b8d2a3

Comment (iakunin) [reply src/main/java/com/artipie/http/rq/RequestLine.java]: @g4s8 thanks

**Comment by iakunin** [comment]
@0crat status

**Comment by iakunin** [comment]
@g4s8 the same situation here: PR is merged, but 0crat didn't close the job

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/artipie/http/issues/13#issuecomment-583258558))

@iakunin This is what I know about this job in [CT2E6TK9B](https://www.0crat.com/p/CT2E6TK9B), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#13](https://github.com/artipie/http/issues/13) is [in scope](http://www.zerocracy.com/policy.html#14) for 3days
  * The role is `REV`
  * The job is assigned to @iakunin[/z](https://www.0crat.com/u/iakunin) for [2days](http://www.zerocracy.com/policy.html#8)
  * There is a monetary reward attached
  * @iakunin[/z](https://www.0crat.com/u/iakunin) will get 25% of their hourly rate for this job
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 15 minutes/points
  * These users are banned and won't be assigned:
    * @g4s8[/z](https://www.0crat.com/u/g4s8): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/CT2E6TK9B?q=%7Bjob%3A%27gh%3Aartipie%2Fhttp%2313%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by iakunin** [comment]
@g4s8 could you, please, do the same as you did [here](https://github.com/artipie/docker-adapter/pull/51#issuecomment-583299501)?

**Comment by 0crat** [comment]
@iakunin[/z](https://www.0crat.com/u/iakunin) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by g4s8** [comment]
@0crat out

**Comment by 0crat** [comment]
Order was finished: +15 point(s) just awarded to @iakunin[/z](https://www.0crat.com/u/iakunin)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Pull request merged**